### PR TITLE
Cypress tests

### PR DESCRIPTION
**Pull request by saqlainrasheed**
# Cases covered
- [X]  anonymous/logged in
- [X] single/ - Multiple submissions
- [X] submit button text
- [X] title
- [X] description
- [X] thank you text
- [X] display: in section
- [X] different input types
- [X] display label
- [X] help text: automatic ("optional" or depending on input type of required)/custom
- [X] required/not required
- [X] validation errors
- [X] default value
- [x] display: embed in the article and direct display
#### polls
- [X] show results/don't show
- [X] results as percentage/total
#### quiz
- [X] correct answer
- [X] feedback text
- [X] multi-page 
#### survey
- [X] multi-page
- [x] survey (skip logic)

**Bug: I have found questions numbering bug on the multi-pages survey. Tests for that particular bug are currently skipped by using ```it.skip()```, You can find them in the ```survey_with_skip_logic.spec.js``` file.**

### CODE REVIEW

File: cypress_tests/cypress/integration/polls/polls_with_logged_in_users.spec.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+describe("Polls for logged in users tests", () => {
+    const url = "/en/sections/questionnaire-testing/poll-only-for-logged-in-users/";
+
+    before("Login the user", () => {
+        cy.login("saqlain", "saqlain");
```
Comment (geoo89) [inline cypress_tests/cypress/integration/polls/polls_with_logged_in_users.spec.js:5]: Could you add another test where no user is logged in, to check that the "Login to participate" button is shown? Note that this was only merged in today, so you might have to merge/rebase onto develop.

File: cypress_tests/cypress/integration/quiz/quiz_with_logged_in_user.spec.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+describe("Quiz with logged in user tests", () => {
+    const url = "/en/sections/questionnaire-testing/sample-quiz/";
+
+    before("Login", () => {
+        cy.login("saqlain", "saqlain");
```
Comment (geoo89) [inline cypress_tests/cypress/integration/quiz/quiz_with_logged_in_user.spec.js:5]: Could you add another test where no user is logged in, to check that the "Login to participate" button is shown? Note that this was only merged in today, so you might have to merge/rebase onto develop.

File: cypress_tests/cypress/integration/survey/survey_with_logged_in_user.spec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+describe("Survey with optional checkbox tests", () => {
+    const url = "/en/sections/questionnaire-testing/survey-with-checkbox/";
+
+    before("Login", () => {
+        cy.login("saqlain", "saqlain");
```
Comment (geoo89) [inline cypress_tests/cypress/integration/survey/survey_with_logged_in_user.spec.js:5]: Could you add another test where no user is logged in, to check that the "Login to participate" button is shown? Note that this was only merged in today, so you might have to merge/rebase onto develop.

File: cypress_tests/cypress/integration/quiz/quiz_with_anonymous_user.spec.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    it("Submits the form and check for not allowed multiple submission", () => {
+        cy.submit(".survey-page__btn>span", "Submit");
+        cy.url().should("include", `/?back_url=${url}&form_length=12`);
+        
+        cy.get(".quiz-answer-banner__counter").contains("7 / 12");
+        cy.get(".quest-item__status").each($el => {
+            cy.wrap($el).contains(/Correct|Incorrect/).should("be.visible");
+        });
+        cy.submit(".icon-btn__title", "Replay Quiz");
+        cy.submit(".survey-page__btn>span", "Submit");
```
Comment (geoo89) [inline cypress_tests/cypress/integration/quiz/quiz_with_anonymous_user.spec.js:48]: A new PR was merged today (so you might have to merge/rebase onto develop) that shows a message saying "YOU HAVE ALREADY COMPLETED THIS SURVEY." now, could you adapt this test please?
Could you also make an equivalent test for polls, where in addition to that text the poll results are shown?

**Review by geoo89 — COMMENTED** [review]
Thanks, this is looking good!

There was a recent PR adding proper views for when you're prevented from submitting because you're not logged in or have already submitted, could you please address the comments I added regarding this?

File: cypress_tests/cypress/integration/polls/polls_with_logged_in_users.spec.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
 describe("Polls for logged in users tests", () => {
     const url = "/en/sections/questionnaire-testing/poll-only-for-logged-in-users/";
 
-    before("Login the user", () => {
+    it("Checks for the login button and login's the user", () => {
```
Comment (ehmadzubair) [inline cypress_tests/cypress/integration/polls/polls_with_logged_in_users.spec.js:4]: logs in

**Review by geoo89 — APPROVED** [review]
This covers all the specified features/options now. (Not all of their combinations, but that's not feasible currently.) I'll will merge this in. Thanks!

**Pull request merged**